### ISSUE TITLE
[COMPATIBILITY]: Legion 7 16ARHA7 Custom Mode

### ISSUE THREAD

**Issue by cameronbosch** [BartoszCichecki/LenovoLegionToolkit#419]
### Version

2.7.0

### OS

Windows 11 22H1

### Device

Legion 7 82UH / 16ARHA7

### Which features of the application do not work as expected?

No option for fan control on Legion 7 2022.

**Comment by BartoszCichecki** [comment]
Which BIOS version do you have?

**Comment by cameronbosch** [comment]
K9CN34WW

**Comment by BartoszCichecki** [comment]
Here is a build that has Custom mode enabled for K9CN bios. Please test it and report back.

[LenovoLegionToolkitSetup_k9cn.zip](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/10127221/LenovoLegionToolkitSetup_k9cn.zip)

**Comment by cameronbosch** [comment]
Where would this option be located?

**Comment by cameronbosch** [comment]
Thank you, I got it working.

**Comment by cameronbosch** [comment]
Although now when I click balance, it switches to Custom Mode.

**Comment by BartoszCichecki** [comment]
Please upload a log whenever you have some problems. You can find instructions on how to get logs in the readme.

**Comment by Rinzler081** [comment]
Let me know if need someone else to test this

**Comment by BartoszCichecki** [comment]
@Rinzler081, yes, you can download the build linked above, do the testing and report back :)

**Comment by henpai91** [comment]
Thanks for making this beta @BartoszCichecki. I was able to test it and can confirm that it's working on my Legion 7 16ARHA7, with a few caveats.

 - It seems to work only when I have my full power AC adapter in (a low wattage adapter) much like performance mode. I can tell I'm on custom mode when my power light turns purple. Do the other 2022 legions require the high power adapter to work?
 
 - When I'm using a low power adapter, I can cycle between Quiet mode and Balance mode using the 'fn+Q' shortcut. When I try to set Balance mode using the Lenovo Legion Toolkit, the setting switches to custom mode after about a second even though the power light on the laptop remains white. I've tested using a custom fan speed when it does this, and the laptop does not use the applied fan settings making me think it really is in Balance mode afterall
